core/domain/user_jobs_one_off.py

```diff
@@ -255,8 +260,17 @@ def map(item):
                 if len(completed_nodes) == len(all_nodes):
                     learner_progress_services.mark_story_as_completed(
                         user_id, story.id)
+                completed_story_ids = (
+                    learner_progress_services.get_all_completed_story_ids(
+                        user_id))
+                story_ids_in_topic = []
+                for canonical_story in topics_linked_to_stories[index].canonical_story_references: # pylint: disable=line-too-long
+                    story_ids_in_topic.append(canonical_story.story_id)
+
+                if (set(story_ids_in_topic).issubset(
+                        set(completed_story_ids))):
```

Comment:
This is wrong, the user just needs to complete **at least one** canonical story, not all of them.

Reply:
Done.